sha1-file.c

```diff
@@ -867,6 +915,24 @@ void prepare_alt_odb(struct repository *r)
 	link_alt_odb_entries(r, r->objects->alternate_db, PATH_SEP, NULL, 0);
 
 	read_info_alternates(r, r->objects->odb->path, 0);
+
+	if (gvfs_shared_cache_pathname && *gvfs_shared_cache_pathname &&
+	    !gvfs_matched_shared_cache_to_alternate) {
+		/*
+		 * There is no entry in .git/objects/info/alternates for
+		 * the shared-cache directory.  Therefore, the odb-list
+		 * does not know about this directory.
+		 *
+		 * Force this directory into the odb-list as an in-memory
+		 * alternate.  Implicitly create the directory on disk, if
+		 * necessary.
+		 *
+		 * See GIT_ALTERNATE_OBJECT_DIRECTORIES for another example
+		 * of this kind of usage.
+		 */
```

Comment:
I think this change is a good one, I'm having trouble thinking of a scenario where we'd have `gvfs.sharedCache` set and not want git to treat it as an alternate.

> Force this directory into the odb-list as an in-memory alternate.

With this change does it still makes sense for us to put `gvfs.sharedCache` in the alternates file when cloning?  Or can/should we rely on git always treating the shared cache location as an alternate?

cc: @derrickstolee for his thoughts as well

Reply:
While this change makes it not _completely necessary_ to have the shared cache in the alternates file, it is still important to put it there, if only for interoperability with other tools or Git versions.

Reply:
Yes, the proper thing to do is to put it into .git/objects/info/alternates during our setup, so that everything
in the Git ecosystem understands that it exists.  This is a backstop to try to keep us from falling back to
.git/objects (and to force the creation of the directory).

I debated updating the .git/objects/info/alternates on disk to completely fix the mis-match, but
chickened out.

Reply:
I updated the comment block to indicate that the don't want to rely on this behavior.